wfe2/wfe.go

```diff
@@ -651,6 +675,23 @@ func (wfe *WebFrontEndImpl) NewAccount(
 			web.ProblemDetailsForError(err, "Error creating new account"), err)
 		return
 	}
+
+	// TODO Andrew: Is this tested?
+	registrationValid := func(reg *corepb.Registration) bool {
+		return !(len(reg.Key) == 0 || len(reg.InitialIP) == 0) && reg.Id != 0
+	}
```

Comment:
I don't think we need to do these checks in the WFE. For InitialIP, we don't actually display that to the user in the generated JSON. For `Key`, it's sufficient to trust that the RA filled it out. This is getting echoed back to the client, and nothing within the WFE depends on the Key being nonempty at this point (though it should be, if everything is working right).

Reply:
The check was being done in the ra client wrapper, which is why it was moved out here. It would save sending it to the ra to check, but I don't have a strong preference to keep it or not.